From bb94591c866684315ba72ad4958a12be4f37918a Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Sun, 2 Oct 2022 12:36:16 +0200
Subject: [PATCH 46/52] Revert "wbt: don't show valid wbt_lat_usec in sysfs
 while wbt is disabled"

This reverts commit 700906e3808fbc8567f826eb7dbb74e2a5f39144.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/blk-sysfs.c | 9 +--------
 block/blk-wbt.c   | 7 -------
 block/blk-wbt.h   | 5 -----
 3 files changed, 1 insertion(+), 20 deletions(-)

diff --git a/block/blk-sysfs.c b/block/blk-sysfs.c
index fca274f15..84d7f8701 100644
--- a/block/blk-sysfs.c
+++ b/block/blk-sysfs.c
@@ -462,14 +462,10 @@ static ssize_t queue_io_timeout_store(struct request_queue *q, const char *page,
 
 static ssize_t queue_wb_lat_show(struct request_queue *q, char *page)
 {
-	u64 lat;
-
 	if (!wbt_rq_qos(q))
 		return -EINVAL;
 
-	lat = wbt_disabled(q) ? 0 : div_u64(wbt_get_min_lat(q), 1000);
-
-	return sprintf(page, "%llu\n", lat);
+	return sprintf(page, "%llu\n", div_u64(wbt_get_min_lat(q), 1000));
 }
 
 static ssize_t queue_wb_lat_store(struct request_queue *q, const char *page,
@@ -492,9 +488,6 @@ static ssize_t queue_wb_lat_store(struct request_queue *q, const char *page,
 			return ret;
 	}
 
-	if (wbt_disabled(q))
-		return -EINVAL;
-
 	if (val == -1)
 		val = wbt_default_latency_nsec(q);
 	else if (val >= 0)
diff --git a/block/blk-wbt.c b/block/blk-wbt.c
index 4c9b3b7af..ae6ea0b54 100644
--- a/block/blk-wbt.c
+++ b/block/blk-wbt.c
@@ -422,13 +422,6 @@ static void wbt_update_limits(struct rq_wb *rwb)
 	rwb_wake_all(rwb);
 }
 
-bool wbt_disabled(struct request_queue *q)
-{
-	struct rq_qos *rqos = wbt_rq_qos(q);
-
-	return !rqos || RQWB(rqos)->enable_state == WBT_STATE_OFF_DEFAULT;
-}
-
 u64 wbt_get_min_lat(struct request_queue *q)
 {
 	struct rq_qos *rqos = wbt_rq_qos(q);
diff --git a/block/blk-wbt.h b/block/blk-wbt.h
index e42465ddc..7e44eccc6 100644
--- a/block/blk-wbt.h
+++ b/block/blk-wbt.h
@@ -94,7 +94,6 @@ void wbt_enable_default(struct request_queue *);
 
 u64 wbt_get_min_lat(struct request_queue *q);
 void wbt_set_min_lat(struct request_queue *q, u64 val);
-bool wbt_disabled(struct request_queue *);
 
 void wbt_set_write_cache(struct request_queue *, bool);
 
@@ -126,10 +125,6 @@ static inline u64 wbt_default_latency_nsec(struct request_queue *q)
 {
 	return 0;
 }
-static inline bool wbt_disabled(struct request_queue *q)
-{
-	return true;
-}
 
 #endif /* CONFIG_BLK_WBT */
 
-- 
2.38.0.rc1.6.g4fd6c5e444

